src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,131 @@
+            while (resultSet.next()) {
+                optionalBook = Optional.of(castResultSetToBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    String.format("Can't find a book with id = %d ", id), e
+            );
+        }
+        return optionalBook;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> list = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement =
+                         connection.prepareStatement(FIND_ALL_QUERY)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                list.add(castResultSetToBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all books from DB", e);
+        }
+        return list;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement =
+                         connection.prepareStatement(UPDATE_QUERY)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            if (statement.executeUpdate() < 1) {
+                throw new DataProcessingException(
+                        String.format(
+                                "Book with id = %d wasn't updated , no rows affected", book.getId()
+                        ));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    String.format(
+                            "Book with id = %d wasn't updated , no rows affected", book.getId()), e
+            );
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement =
+                         connection.prepareStatement(DELETE_BY_ID_QUERY)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    String.format("Book with id = %d wasn't delete", id), e);
+        }
+    }
+
+    private Long getGeneratedId(PreparedStatement preparedStatement) throws SQLException {
+        try (ResultSet generatedKeys = preparedStatement.getGeneratedKeys()) {
+            if (generatedKeys.next()) {
+                return generatedKeys.getObject(1, Long.class);
+            }
+        }
+        throw new DataProcessingException("Can't get generated id ");
+    }
+
+    private Book castResultSetToBook(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getObject("id", Long.class);
```

Comment:
since you're throwing exception during insert if id wasn't generated - you can use `getLong` here

at this stage id can't be null due to previous checks